From 9c977927e545943996981c86c7ec71f0b44921ba Mon Sep 17 00:00:00 2001
From: Sebastian Huber <sebastian.huber@embedded-brains.de>
Date: Sun, 17 Jul 2011 15:14:40 +0200
Subject: [PATCH 4/6] Typo.

---
 hw/arm_gic.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/arm_gic.c b/hw/arm_gic.c
index a97a318..237f13f 100644
--- a/hw/arm_gic.c
+++ b/hw/arm_gic.c
@@ -378,7 +378,7 @@ static uint32_t gic_dist_readl(void *opaque, target_phys_addr_t offset)
     uint32_t addr;
     addr = offset;
     if (addr < 0x100 || (addr > 0xd00 && addr != 0xd18 && addr != 0xd1c
-        && addr != 0xd20)) {
+        && addr != 0xd20))
         return nvic_readl(s, addr);
 #endif
     val = gic_dist_readw(opaque, offset);
-- 
1.7.1

