diff -uNr binutils-2.17/ld/configure.tgt /home2/distros/RH72/home/rtems/src/rtems/contrib//crossrpms/rtems4.7/binutils-2.17/ld/configure.tgt
--- binutils-2.17/ld/configure.tgt	2006-04-05 07:41:57.000000000 -0500
+++ /home2/distros/RH72/home/rtems/src/rtems/contrib//crossrpms/rtems4.7/binutils-2.17/ld/configure.tgt	2006-08-10 13:03:26.000000000 -0500
@@ -83,7 +83,8 @@
 avr-*-*)		targ_emul=avr2
 			targ_extra_emuls="avr1 avr3 avr4 avr5"
 			;;
-bfin-*-elf)		targ_emul=elf32bfin; targ_extra_emuls="elf32bfinfd" ;;
+bfin-*-elf | bfin-*-rtems*)
+			targ_emul=elf32bfin; targ_extra_emuls="elf32bfinfd" ;;
 bfin-*-uclinux*)	targ_emul=elf32bfin; targ_extra_emuls="elf32bfinfd" ;;
 cr16c-*-elf*)           targ_emul=elf32cr16c
 			;;
