diff -uNr binutils-2.17.50.orig/ld/configure.tgt binutils-2.17.50/ld/configure.tgt
--- binutils-2.17.50.orig/ld/configure.tgt	2007-04-04 13:02:39.000000000 -0500
+++ binutils-2.17.50/ld/configure.tgt	2007-05-25 05:17:14.000000000 -0500
@@ -113,6 +113,10 @@
 			targ_extra_emuls="elf32bfinfd"
 			targ_extra_libpath=$targ_extra_emuls
 			;;
+bfin-*-rtems*)		targ_emul=elf32bfin;
+			targ_extra_emuls="elf32bfinfd"
+			targ_extra_libpath=$targ_extra_emuls
+			;;
 bfin-*-uclinux*)	targ_emul=elf32bfin;
 			targ_extra_emuls="elf32bfinfd"
 			targ_extra_libpath=$targ_extra_emuls
