diff -uNr binutils-2.17.90.orig/ld/configure.tgt binutils-2.17.90/ld/configure.tgt
--- binutils-2.17.90.orig/ld/configure.tgt	2007-08-06 22:00:19.000000000 +0200
+++ binutils-2.17.90/ld/configure.tgt	2007-08-07 08:42:24.000000000 +0200
@@ -113,6 +113,10 @@
 			targ_extra_emuls="elf32bfinfd"
 			targ_extra_libpath=$targ_extra_emuls
 			;;
+bfin-*-rtems*)		targ_emul=elf32bfin;
+			targ_extra_emuls="elf32bfinfd"
+			targ_extra_libpath=$targ_extra_emuls
+			;;
 bfin-*-uclinux*)	targ_emul=elf32bfin;
 			targ_extra_emuls="elf32bfinfd"
 			targ_extra_libpath=$targ_extra_emuls
