diff -uNr gcc-4.2.2.orig/gcc/config/arm/rtems-elf.h gcc-4.2.2/gcc/config/arm/rtems-elf.h
--- gcc-4.2.2.orig/gcc/config/arm/rtems-elf.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.2/gcc/config/arm/rtems-elf.h	2007-11-27 18:50:47.000000000 +0100
@@ -26,6 +26,7 @@
 #define TARGET_OS_CPP_BUILTINS()		\
     do {					\
 	builtin_define ("__rtems__");		\
+	builtin_define ("__USE_INIT_FINI__");	\
 	builtin_assert ("system=rtems");	\
     } while (0)
 
diff -uNr gcc-4.2.2.orig/gcc/config/c4x/rtems.h gcc-4.2.2/gcc/config/c4x/rtems.h
--- gcc-4.2.2.orig/gcc/config/c4x/rtems.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.2/gcc/config/c4x/rtems.h	2007-11-27 18:50:47.000000000 +0100
@@ -23,6 +23,5 @@
 #define TARGET_OS_CPP_BUILTINS()		\
     do {					\
 	builtin_define ("__rtems__");		\
-	builtin_define ("__USE_INIT_FINI__");	\
 	builtin_assert ("system=rtems");	\
     } while (0)
diff -uNr gcc-4.2.2.orig/gcc/config/m68k/t-rtems gcc-4.2.2/gcc/config/m68k/t-rtems
--- gcc-4.2.2.orig/gcc/config/m68k/t-rtems	2005-02-07 06:13:25.000000000 +0100
+++ gcc-4.2.2/gcc/config/m68k/t-rtems	2007-11-27 18:50:47.000000000 +0100
@@ -1,7 +1,14 @@
 # Custom multilibs for RTEMS
 
-MULTILIB_OPTIONS = m68000/m68020/m5200/mcpu32/m68030/m68040/m68060 m68881/msoft-float
+MULTILIB_OPTIONS = m68000/m68020/m5200/m528x/mcpu32/m68030/m68040/m68060/m5307/m5407/mcfv4e
+MULTILIB_OPTIONS += m68881/msoft-float
 MULTILIB_DIRNAMES =
 MULTILIB_MATCHES = m68000=mc68000 m68000=m68302 mcpu32=m68332 m68020=mc68020 m68030=mc68030
-MULTILIB_MATCHES += m5200=m528x
-MULTILIB_EXCEPTIONS = m68000/msoft-float m5200/m68881 m5200/msoft-float mcpu32/m68881 mcpu32/msoft-float m68040/m68881 m68060/m68881
+MULTILIB_MATCHES += m5200=m5206e
+MULTILIB_EXCEPTIONS = m68000/msoft-float m68040/m68881 m68060/m68881
+MULTILIB_EXCEPTIONS += mcpu32/m68881 mcpu32/msoft-float
+MULTILIB_EXCEPTIONS += m5200/m68881 m5200/msoft-float
+MULTILIB_EXCEPTIONS += m528x/m68881 m528x/msoft-float
+MULTILIB_EXCEPTIONS += m5307/m68881 m5307/msoft-float
+MULTILIB_EXCEPTIONS += m5407/m68881 m5407/msoft-float
+MULTILIB_EXCEPTIONS += mcfv4e/m68881 mcfv4e/msoft-float
diff -uNr gcc-4.2.2.orig/gcc/config/mips/elf.h gcc-4.2.2/gcc/config/mips/elf.h
--- gcc-4.2.2.orig/gcc/config/mips/elf.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.2/gcc/config/mips/elf.h	2007-11-27 18:50:47.000000000 +0100
@@ -48,6 +48,4 @@
 #undef  ENDFILE_SPEC
 #define ENDFILE_SPEC "crtend%O%s crtn%O%s"
 
-#define NO_IMPLICIT_EXTERN_C 1
-
 #define HANDLE_PRAGMA_PACK_PUSH_POP 1
diff -uNr gcc-4.2.2.orig/gcc/gthr-rtems.h gcc-4.2.2/gcc/gthr-rtems.h
--- gcc-4.2.2.orig/gcc/gthr-rtems.h	2005-06-25 04:02:01.000000000 +0200
+++ gcc-4.2.2/gcc/gthr-rtems.h	2007-11-28 04:39:49.000000000 +0100
@@ -38,7 +38,6 @@
 #define __GTHREADS 1
 
 #define __GTHREAD_ONCE_INIT  0
-#define __GTHREAD_MUTEX_INIT 0
 #define __GTHREAD_MUTEX_INIT_FUNCTION  rtems_gxx_mutex_init
 #define __GTHREAD_RECURSIVE_MUTEX_INIT_FUNCTION  rtems_gxx_recursive_mutex_init
 
