diff -urN gcc-4.4.2-orig/gcc/ada/s-osinte-rtems.ads gcc-4.4.2/gcc/ada/s-osinte-rtems.ads
--- gcc-4.4.2-orig/gcc/ada/s-osinte-rtems.ads	2009-04-09 18:23:07.000000000 -0500
+++ gcc-4.4.2/gcc/ada/s-osinte-rtems.ads	2009-11-30 12:03:37.000000000 -0600
@@ -625,6 +625,7 @@
       process_shared  : int;
       prio_ceiling    : int;
       protocol        : int;
+      mutex_type      : int;
       recursive       : int;
    end record;
    pragma Convention (C, pthread_mutexattr_t);
diff -urN gcc-4.4.2-orig/gcc/ada/s-stchop-rtems.adb gcc-4.4.2/gcc/ada/s-stchop-rtems.adb
--- gcc-4.4.2-orig/gcc/ada/s-stchop-rtems.adb	2009-04-09 18:23:07.000000000 -0500
+++ gcc-4.4.2/gcc/ada/s-stchop-rtems.adb	2009-11-30 12:03:37.000000000 -0600
@@ -80,8 +80,9 @@
    is
       pragma Unreferenced (Stack_Address);
 
-      --  RTEMS has a routine to check this.  So use it.
-      function rtems_stack_checker_is_blown return Interfaces.C.int;
+      --  RTEMS has a routine to check if the stack is blown.
+      --  It returns a C99 bool.
+      function rtems_stack_checker_is_blown return Interfaces.C.unsigned_char;
       pragma Import (C,
          rtems_stack_checker_is_blown, "rtems_stack_checker_is_blown");
 
