--- /home/joel/tools-original/gcc-4.2.0/gcc/ada/s-osinte-rtems.ads	2005-06-30 20:29:17.000000000 -0500
+++ gcc-4.2.0/gcc/ada/s-osinte-rtems.ads	2007-06-22 17:25:56.000000000 -0500
@@ -291,12 +291,10 @@
       sig    : Signal) return int;
    pragma Import (C, pthread_kill, "pthread_kill");
 
-   type sigset_t_ptr is access all sigset_t;
-
    function pthread_sigmask
      (how  : int;
-      set  : sigset_t_ptr;
-      oset : sigset_t_ptr) return int;
+      set  : access sigset_t;
+      oset : access sigset_t) return int;
    pragma Import (C, pthread_sigmask, "pthread_sigmask");
 
    ----------------------------
