diff -Naur gcc-4.2.4.orig/configure gcc-4.2.4/configure
--- gcc-4.2.4.orig/configure	2007-09-15 02:42:24.000000000 +0200
+++ gcc-4.2.4/configure	2008-05-26 08:48:31.000000000 +0200
@@ -1260,7 +1260,7 @@
     noconfigdirs="$noconfigdirs target-newlib target-libiberty target-libgloss ${libgcj} target-libmudflap"
     ;;
   *-*-rtems*)
-    noconfigdirs="$noconfigdirs target-libgloss ${libgcj}"
+    noconfigdirs="$noconfigdirs target-libiberty target-libgloss ${libgcj}"
     ;;
     # The tpf target doesn't support gdb yet.
   *-*-tpf*)
diff -Naur gcc-4.2.4.orig/configure.in gcc-4.2.4/configure.in
--- gcc-4.2.4.orig/configure.in	2007-09-15 02:42:24.000000000 +0200
+++ gcc-4.2.4/configure.in	2008-05-26 08:48:31.000000000 +0200
@@ -420,7 +420,7 @@
     noconfigdirs="$noconfigdirs target-newlib target-libiberty target-libgloss ${libgcj} target-libmudflap"
     ;;
   *-*-rtems*)
-    noconfigdirs="$noconfigdirs target-libgloss ${libgcj}"
+    noconfigdirs="$noconfigdirs target-libiberty target-libgloss ${libgcj}"
     ;;
     # The tpf target doesn't support gdb yet.
   *-*-tpf*)
diff -Naur gcc-4.2.4.orig/gcc/config/c4x/rtems.h gcc-4.2.4/gcc/config/c4x/rtems.h
--- gcc-4.2.4.orig/gcc/config/c4x/rtems.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.4/gcc/config/c4x/rtems.h	2008-05-26 08:51:24.000000000 +0200
@@ -23,6 +23,5 @@
 #define TARGET_OS_CPP_BUILTINS()		\
     do {					\
 	builtin_define ("__rtems__");		\
-	builtin_define ("__USE_INIT_FINI__");	\
 	builtin_assert ("system=rtems");	\
     } while (0)
diff -Naur gcc-4.2.4.orig/gcc/config/mips/elf.h gcc-4.2.4/gcc/config/mips/elf.h
--- gcc-4.2.4.orig/gcc/config/mips/elf.h	2007-09-01 17:28:30.000000000 +0200
+++ gcc-4.2.4/gcc/config/mips/elf.h	2008-05-26 08:58:40.000000000 +0200
@@ -48,6 +48,4 @@
 #undef  ENDFILE_SPEC
 #define ENDFILE_SPEC "crtend%O%s crtn%O%s"
 
-#define NO_IMPLICIT_EXTERN_C 1
-
 #define HANDLE_PRAGMA_PACK_PUSH_POP 1
