diff -uNr gcc-4.2.1.orig/configure gcc-4.2.1/configure
--- gcc-4.2.1.orig/configure	2007-06-21 01:07:21.000000000 +0200
+++ gcc-4.2.1/configure	2007-11-29 14:28:13.000000000 +0100
@@ -3743,7 +3743,7 @@
     # For an installed makeinfo, we require it to be from texinfo 4.4 or
     # higher, else we use the "missing" dummy.
     if ${MAKEINFO} --version \
-       | egrep 'texinfo[^0-9]*([1-3][0-9]|4\.[4-9]|[5-9])' >/dev/null 2>&1; then
+       | egrep 'texinfo[^0-9]*(4\.([4-9]|[1-9][0-9])|[5-9]|[1-9][0-9])' >/dev/null 2>&1; then
       :
     else
       MAKEINFO="$MISSING makeinfo"
diff -uNr gcc-4.2.1.orig/configure.in gcc-4.2.1/configure.in
--- gcc-4.2.1.orig/configure.in	2007-05-30 15:48:07.000000000 +0200
+++ gcc-4.2.1/configure.in	2007-11-29 14:28:13.000000000 +0100
@@ -2246,7 +2246,7 @@
     # For an installed makeinfo, we require it to be from texinfo 4.4 or
     # higher, else we use the "missing" dummy.
     if ${MAKEINFO} --version \
-       | egrep 'texinfo[^0-9]*([1-3][0-9]|4\.[4-9]|[5-9])' >/dev/null 2>&1; then
+       | egrep 'texinfo[^0-9]*(4\.([4-9]|[1-9][0-9])|[5-9]|[1-9][0-9])' >/dev/null 2>&1; then
       :
     else
       MAKEINFO="$MISSING makeinfo"
diff -uNr gcc-4.2.1.orig/gcc/config/arm/rtems-elf.h gcc-4.2.1/gcc/config/arm/rtems-elf.h
--- gcc-4.2.1.orig/gcc/config/arm/rtems-elf.h	2005-11-21 23:56:34.000000000 +0100
+++ gcc-4.2.1/gcc/config/arm/rtems-elf.h	2007-11-29 14:28:13.000000000 +0100
@@ -27,6 +27,7 @@
 #define TARGET_OS_CPP_BUILTINS()		\
     do {					\
 	builtin_define ("__rtems__");		\
+	builtin_define ("__USE_INIT_FINI__");	\
 	builtin_assert ("system=rtems");	\
     } while (0)
 
diff -uNr gcc-4.2.1.orig/gcc/config/c4x/rtems.h gcc-4.2.1/gcc/config/c4x/rtems.h
--- gcc-4.2.1.orig/gcc/config/c4x/rtems.h	2005-06-25 03:22:41.000000000 +0200
+++ gcc-4.2.1/gcc/config/c4x/rtems.h	2007-11-29 14:28:13.000000000 +0100
@@ -24,6 +24,5 @@
 #define TARGET_OS_CPP_BUILTINS()		\
     do {					\
 	builtin_define ("__rtems__");		\
-	builtin_define ("__USE_INIT_FINI__");	\
 	builtin_assert ("system=rtems");	\
     } while (0)
diff -uNr gcc-4.2.1.orig/gcc/config/m68k/t-rtems gcc-4.2.1/gcc/config/m68k/t-rtems
--- gcc-4.2.1.orig/gcc/config/m68k/t-rtems	2005-02-07 06:13:25.000000000 +0100
+++ gcc-4.2.1/gcc/config/m68k/t-rtems	2007-11-29 14:30:08.000000000 +0100
@@ -1,7 +1,14 @@
 # Custom multilibs for RTEMS
 
-MULTILIB_OPTIONS = m68000/m68020/m5200/mcpu32/m68030/m68040/m68060 m68881/msoft-float
+MULTILIB_OPTIONS = m68000/m68020/m5200/m528x/mcpu32/m68030/m68040/m68060/m5307/m5407/mcfv4e
+MULTILIB_OPTIONS += m68881/msoft-float
 MULTILIB_DIRNAMES =
 MULTILIB_MATCHES = m68000=mc68000 m68000=m68302 mcpu32=m68332 m68020=mc68020 m68030=mc68030
-MULTILIB_MATCHES += m5200=m528x
-MULTILIB_EXCEPTIONS = m68000/msoft-float m5200/m68881 m5200/msoft-float mcpu32/m68881 mcpu32/msoft-float m68040/m68881 m68060/m68881
+MULTILIB_MATCHES += m5200=m5206e
+MULTILIB_EXCEPTIONS = m68000/msoft-float m68040/m68881 m68060/m68881
+MULTILIB_EXCEPTIONS += mcpu32/m68881 mcpu32/msoft-float
+MULTILIB_EXCEPTIONS += m5200/m68881 m5200/msoft-float
+MULTILIB_EXCEPTIONS += m528x/m68881 m528x/msoft-float
+MULTILIB_EXCEPTIONS += m5307/m68881 m5307/msoft-float
+MULTILIB_EXCEPTIONS += m5407/m68881 m5407/msoft-float
+MULTILIB_EXCEPTIONS += mcfv4e/m68881 mcfv4e/msoft-float
diff -uNr gcc-4.2.1.orig/gcc/config/mips/elf.h gcc-4.2.1/gcc/config/mips/elf.h
--- gcc-4.2.1.orig/gcc/config/mips/elf.h	2006-10-08 09:35:47.000000000 +0200
+++ gcc-4.2.1/gcc/config/mips/elf.h	2007-11-29 14:28:13.000000000 +0100
@@ -49,6 +49,4 @@
 #undef  ENDFILE_SPEC
 #define ENDFILE_SPEC "crtend%O%s crtn%O%s"
 
-#define NO_IMPLICIT_EXTERN_C 1
-
 #define HANDLE_PRAGMA_PACK_PUSH_POP 1
diff -uNr gcc-4.2.1.orig/gcc/configure gcc-4.2.1/gcc/configure
--- gcc-4.2.1.orig/gcc/configure	2007-01-02 04:44:31.000000000 +0100
+++ gcc-4.2.1/gcc/configure	2007-11-29 14:28:13.000000000 +0100
@@ -7654,7 +7654,7 @@
   echo "configure:7654: version of makeinfo is $ac_prog_version" >&5
   case $ac_prog_version in
     '')     gcc_cv_prog_makeinfo_modern=no;;
-    4.[4-9]*)
+    4.[4-9]*|4.[1-9][0-9]*|[5-9]*|[1-9][0-9]*)
             gcc_cv_prog_makeinfo_modern=yes;;
     *)      gcc_cv_prog_makeinfo_modern=no;;
   esac
@@ -13937,9 +13937,7 @@
   ld_ver=`$gcc_cv_ld --version 2>/dev/null | sed 1q`
   if echo "$ld_ver" | grep GNU > /dev/null; then
     ld_vers=`echo $ld_ver | sed -n \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
+	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*.*\)$,\1,p' \
 	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\)[	 ].*$,\1,p' \
 	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)[	 ].*$,\1,p' \
 	-e 's,^.*[ 	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)[ 	].*$,\1,p' \
diff -uNr gcc-4.2.1.orig/gcc/configure.ac gcc-4.2.1/gcc/configure.ac
--- gcc-4.2.1.orig/gcc/configure.ac	2007-01-02 04:44:31.000000000 +0100
+++ gcc-4.2.1/gcc/configure.ac	2007-11-29 14:28:13.000000000 +0100
@@ -878,7 +878,7 @@
 # that we can use it.
 gcc_AC_CHECK_PROG_VER(MAKEINFO, makeinfo, --version,
   [GNU texinfo.* \([0-9][0-9.]*\)],
-  [4.[4-9]*])
+  [4.[4-9]*|4.[1-9][0-9]*|[5-9]*|[1-9][0-9]*])
 if test $gcc_cv_prog_makeinfo_modern = no; then
   MAKEINFO="$MISSING makeinfo"
   AC_MSG_WARN([
@@ -2094,9 +2094,7 @@
   if echo "$ld_ver" | grep GNU > /dev/null; then
 changequote(,)dnl
     ld_vers=`echo $ld_ver | sed -n \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
-	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)$,\1,p' \
+	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*.*\)$,\1,p' \
 	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\)[	 ].*$,\1,p' \
 	-e 's,^.*[	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)[	 ].*$,\1,p' \
 	-e 's,^.*[ 	 ]\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)[ 	].*$,\1,p' \
diff -uNr gcc-4.2.1.orig/gcc/gthr-rtems.h gcc-4.2.1/gcc/gthr-rtems.h
--- gcc-4.2.1.orig/gcc/gthr-rtems.h	2005-06-25 04:02:01.000000000 +0200
+++ gcc-4.2.1/gcc/gthr-rtems.h	2007-11-29 14:30:08.000000000 +0100
@@ -38,7 +38,6 @@
 #define __GTHREADS 1
 
 #define __GTHREAD_ONCE_INIT  0
-#define __GTHREAD_MUTEX_INIT 0
 #define __GTHREAD_MUTEX_INIT_FUNCTION  rtems_gxx_mutex_init
 #define __GTHREAD_RECURSIVE_MUTEX_INIT_FUNCTION  rtems_gxx_recursive_mutex_init
 
