--- gdb-6.4/sim/Makefile.in	29 Jan 2005 00:53:13 -0000	1.6
+++ gdb-6.4/sim/Makefile.in	11 Mar 2006 04:59:44 -0000
@@ -87,6 +87,7 @@
 	"exec_prefix=$(exec_prefix)" \
 	"bindir=$(bindir)" \
 	"mandir=$(mandir)" \
+	"libdir=$(libdir)" \
 	"against=$(against)" \
 	"AR=$(AR)" \
 	"AR_FLAGS=$(AR_FLAGS)" \
--- gdb-6.4/sim/erc32/exec.c	7 Mar 2005 11:09:05 -0000	1.2
+++ gdb-6.4/sim/erc32/exec.c	11 Mar 2006 04:59:45 -0000
@@ -1713,7 +1713,7 @@
     	    sregs->fdp[rs2 | 1] = sregs->fs[rs2 & ~1];
     	    sregs->fdp[rs2 & ~1] = sregs->fs[rs2 | 1];
     default:
-      ;
+      break;
     }
 #endif
 
@@ -1886,7 +1886,7 @@
 	sregs->fs[rd & ~1] = sregs->fdp[rd | 1];
 	sregs->fs[rd | 1] = sregs->fdp[rd & ~1];
     default:
-      ;
+      break;
     }
 #endif
     if (sregs->fpstate == FP_EXC_PE) {
--- gdb-6.4/sim/mips/dv-tx3904tmr.c	16 Apr 1999 01:35:06 -0000	1.1.1.1
+++ gdb-6.4/sim/mips/dv-tx3904tmr.c	11 Mar 2006 04:59:46 -0000
@@ -675,6 +675,7 @@
 
 	case 3: /* disabled */
 	default:
+	  break;
 	}
 
       /* update counter and report */
